feature_engine/creation/decision_tree.py

```diff
@@ -0,0 +1,328 @@
+        to the number of provided variables when the class is initiated.
+
+    regression: boolean, default = True
+        Select whether the decision tree is performing a regression or
+        classification.
+
+    max_depth: integer, default = True
+        The maximum depth of the tree. Used to mitigate the risk of over-fitting.
+
+    {missing_values}
+
+    {drop_original}
+
+    Attributes
+    ----------
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    fit:
+        Builds a decision tree estimator(s).
+
+    {transform}
+
+    {fit_transform}
+
+    Notes
+    -----
+
+
+    """
+    def __init__(
+        self,
+        variables: List[Union[str, int]] = None,
+        # TODO: What is the correct typing hint?
+        output_features: Union[int, List[int], Tuple[tuple, ...]] = None,
+        regression: bool = True,
+        max_depth: int = 3,
+        missing_value: str = "raise",
+        drop_original: bool = False,
+    ) -> None:
+
+        if (
+            not isinstance(variables, list)
+            or not all(isinstance(var, (int, str)) for var in variables)
+            or len(set(variables)) != len(variables)
+        ):
+            raise ValueError(
+                "variables must a list of strings or integers comprise of "
+                f"distinct variables. Got {variables} instead."
+            )
+
+        if (
+                not isinstance(output_features, (int, list, tuple))
+                and output_features is not None
+        ):
+            raise ValueError(
+                f"output_features must an integer, list or tuple. Got "
+                f"{output_features} instead."
+            )
+
+        if isinstance(output_features, int):
+            if output_features > len(variables):
+                raise ValueError(
+                    "If output_features is an integer, the value cannot be "
+                    f"greater than the length of variables. Got {output_features} "
+                    "for output_features and {len(variables)} for the length "
+                    "of variables."
+                )
+
+        if isinstance(output_features, list):
+            if (
+                    max(output_features) > len(variables)
+                    or not all(
+                            isinstance(feature, int) for feature in output_features
+                        )
+            ):
+                raise ValueError(
+                    "output_features must be a list solely comprised of integers "
+                    "and the maximum integer cannot be greater than the length of "
+                    f"variables. Got {output_features} for output_features and "
+                )
+
+        if isinstance(output_features, tuple):
+            num_combos = 0
+            for n in range(1, len(variables) + 1):
+                num_combos += len(list(combinations(variables, n)))
```

Comment:
could we add a small comment to explain what this is doing? I am so not familiar with itertools, it would help me.